Code/Framework/AzFramework/AzFramework/Spawnable/SpawnableMediator.cpp

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzFramework/Components/TransformComponent.h>
+#include <AzFramework/Spawnable/SpawnableBus.h>
+#include <AzFramework/Spawnable/SpawnableMediator.h>
+#include <AzFramework/Spawnable/SpawnableNotificationsHandler.h>
+
+namespace AzFramework
+{
+    void SpawnableMediator::Reflect(AZ::ReflectContext* context)
+    {
+        SpawnableAssetRef::Reflect(context);
+
+        if (auto* serializeContext = azrtti_cast<AZ::SerializeContext*>(context); serializeContext != nullptr)
+        {
+            serializeContext->Class<SpawnableMediator>();
+        }
+
+        if (auto behaviorContext = azrtti_cast<AZ::BehaviorContext*>(context))
+        {
+            behaviorContext->Class<SpawnableMediator>("SpawnableMediator")
+                ->Attribute(AZ::Script::Attributes::Scope, AZ::Script::Attributes::ScopeFlags::Common)
+                ->Attribute(AZ::Script::Attributes::Category, "Prefab/Spawning")
+                ->Attribute(AZ::Script::Attributes::Module, "prefabs")
+                ->Method("CreateSpawnTicket", &SpawnableMediator::CreateSpawnTicket)
+                ->Method("Spawn", &SpawnableMediator::Spawn)
+                ->Method("Despawn", &SpawnableMediator::Despawn)
+                ->Method("Clear", &SpawnableMediator::Clear);
+
+            behaviorContext->EBus<SpawnableNotificationsBus>("SpawnableNotificationsBus")
+                ->Attribute(AZ::Script::Attributes::Scope, AZ::Script::Attributes::ScopeFlags::Common)
+                ->Attribute(AZ::Script::Attributes::Category, "Prefab/Spawning")
+                ->Attribute(AZ::Script::Attributes::Module, "prefabs")
+                ->Handler<SpawnableNotificationsHandler>();
+        }
+    }
+
+    SpawnableMediator::SpawnableMediator()
+        : m_interface(SpawnableEntitiesInterface::Get())
+    {
+    }
+
+    SpawnableMediator::~SpawnableMediator()
+    {
+        Clear();
+    }
+
+    void SpawnableMediator::OnTick(float /*deltaTime*/, AZ::ScriptTimePoint /*time*/)
+    {
+        ProcessDespawnedResults();
+        ProcessSpawnedResults();
+        AZ::TickBus::Handler::BusDisconnect();
+    }
+
+    EntitySpawnTicket SpawnableMediator::CreateSpawnTicket(const SpawnableAssetRef& spawnableAsset)
+    {
+        return EntitySpawnTicket(spawnableAsset.m_asset);
+    }
+
+    bool SpawnableMediator::Spawn(
+        EntitySpawnTicket spawnTicket,
+        AZ::EntityId parentId,
+        AZ::Vector3 translation,
+        AZ::Vector3 rotation,
+        float scale)
```

Comment:
```suggestion
        const EntitySpawnTicket spawnTicket,
        const AZ::EntityId& parentId,
        const AZ::Vector3& translation,
        const AZ::Vector3& rotation,
        float scale)
```

Reply:
can't have const ticket, but updated other fields. Should EntityId also be const &?